packages/cli/src/ui/hooks/usePermissionsModifyTrust.ts

```diff
@@ -0,0 +1,119 @@
+/**
+ * @license
+ * Copyright 2025 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { useState, useEffect, useCallback } from 'react';
+import * as process from 'node:process';
+import {
+  loadTrustedFolders,
+  TrustLevel,
+  isWorkspaceTrusted,
+} from '../../config/trustedFolders.js';
+import { useSettings } from '../contexts/SettingsContext.js';
+
+import { MessageType } from '../types.js';
+import { type UseHistoryManagerReturn } from './useHistoryManager.js';
+
+export const usePermissionsModifyTrust = (
+  onExit: () => void,
+  addItem: UseHistoryManagerReturn['addItem'],
+) => {
+  const [loading, setLoading] = useState(true);
+  const [currentTrustLevel, setCurrentTrustLevel] = useState<
+    TrustLevel | undefined
+  >();
+  const [pendingTrustLevel, setPendingTrustLevel] = useState<
+    TrustLevel | undefined
+  >();
+  const [isInheritedTrustFromParent, setIsInheritedTrustFromParent] =
+    useState(false);
+  const [isInheritedTrustFromIde, setIsInheritedTrustFromIde] = useState(false);
+  const [needsRestart, setNeedsRestart] = useState(false);
+  const settings = useSettings();
+  const cwd = process.cwd();
+
+  const isFolderTrustEnabled = !!settings.merged.security?.folderTrust?.enabled;
+
+  useEffect(() => {
+    if (!isFolderTrustEnabled) {
+      setLoading(false);
+      return;
+    }
+    const folders = loadTrustedFolders();
+    const explicitTrustLevel = folders.user.config[cwd];
+    setCurrentTrustLevel(explicitTrustLevel);
+
+    const { isTrusted, source } = isWorkspaceTrusted(settings.merged);
+
+    const isInheritedTrust =
+      isTrusted &&
+      (!explicitTrustLevel || explicitTrustLevel === TrustLevel.DO_NOT_TRUST);
+
+    setIsInheritedTrustFromIde(!!(source === 'ide' && isInheritedTrust));
+    setIsInheritedTrustFromParent(!!(source === 'file' && isInheritedTrust));
+
+    setLoading(false);
+  }, [cwd, settings.merged, isFolderTrustEnabled]);
```

Comment:
why are we using useEffect? Shouldn't we just compute isInheritedTrustFromIde based on the source and isInhertedTrust?